Memory_Management/Paging.md

```diff
@@ -13,13 +74,91 @@ To implement paging, is strongly reccomended to have implemented exceptions too.
 The 4 levels of page directories/tables are: 
 
 * the Page-Map Level-4 Table (PML4),
-* the Page-Directory Pointer Table (PDP),
+* the Page-Directory Pointer Table (PDPR),
 * the Page-Directory Table (PD),
 * and the Page Table (PT).
 
 The number of levels depend on the size of the pages chosen. 
 If we are using 4kb pages then we will have: PML4, PDPR, PD, PT, while if we go for 2mb Pages we have only PML4, PDPR, PD. 
 
+## Page Directories and Table structure
+
+As we have seen earlier in this section, when paging is enabled, a virtual address is translated into a set of entry number in different tables. In this paragraph we will see the different types available for them
+
+But before proceeding with the details let's see some of the characteristics common between all table/directory types: 
+
+* The size of all table type is fixed and is 4k
+* Every table has exactly 512 entries
+* Every entry has the size of 64 bits
+* The tables have a hierarchy, and every item in a table higher in the hierachy point to a lower hierachy one (with some exceptions explained later). The page table points to a memory area. 
+
+The hierarchy of the tables is: 
+
+* PML4 is the Root table, is the one that is contained in the PDBR register, and that is loaded for the actual address translation (see the next paragraph). Every entry in the page table points to a PDPR table
+* PDPR every entry of this table points to a Page Directory
```

Comment:
worth mentioning that if a level 3 page has the PS bit set, its a 1gb page. This feature is not supported by all CPUs though, and cannot be safely assumed like NX or x2apic.